src/hotspot/share/interpreter/zero/bytecodeInterpreter.cpp

```diff
@@ -403,25 +403,25 @@ template void BytecodeInterpreter::run<false>(interpreterState istate);
 
 template<bool JVMTI_ENABLED>
 void BytecodeInterpreter::run(interpreterState istate) {
-
-  // In order to simplify some tests based on switches set at runtime
-  // we invoke the interpreter a single time after switches are enabled
-  // and set simpler to to test variables rather than method calls or complex
-  // boolean expressions.
-
-  static int initialized = 0;
-  static int checkit = 0;
-  static intptr_t* c_addr = NULL;
-  static intptr_t  c_value;
-
-  if (checkit && *c_addr != c_value) {
-    os::breakpoint();
-  }
+  intptr_t*        topOfStack = (intptr_t *)istate->stack(); /* access with STACK macros */
+  address          pc = istate->bcp();
+  jubyte opcode;
+  intptr_t*        locals = istate->locals();
+  ConstantPoolCache*    cp = istate->constants(); // method()->constants()->cache()
+#ifdef LOTS_OF_REGS
+  JavaThread*      THREAD = istate->thread();
+#else
+#undef THREAD
+#define THREAD istate->thread()
+#endif
 
```

Comment:
This is a weirdly-hacky optimization, and is perhaps obsolete on modern compilers. While simplifying, I'd take it out.

Reply:
I remember following up on this whole `LOTS_OF_REGS` mess, and it seems still profitable. I can take a look in a separate RFE, OK?

Reply:
OK.